test/TestApp.AspNetCore.3.1/ActivityMiddleware.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="ActivityMiddleware.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace TestApp.AspNetCore._3._1
+{
+    public class ActivityMiddleware
```

Comment:
It's probably a good idea to create a shared library under `test/` to keep these helper classes such as `ActivityMiddleware` and `CallbackMiddleware` and have these two projects refer to the shared library.

Reply:
Yes - planning to do this after .NET7.0 branch. Could use the same app for multiple targets as well.